File: Seccion 12 - Punteros en C++/Seccion_12_Dart/02_ejercicio_02.dart

import 'dart:io';

void main() {

  int numero, cont = 0;

  stdout.write('Digite un número: ');
  numero = int.parse(stdin.readLineSync());

  for (int i = 1; i < numero; i++) {
    if (numero % i == 0) {
      cont++;
    }
  }

  if (cont > 2) {
    print('El número $numero no es primo.');
  } else {
    print('El número $numero es primo.');
  }

}

File: Seccion 12 - Punteros en C++/Seccion_12_Python/02_ejercicio_02.py

def main():
    
    numero: int
    cont: int = 0
    
    numero = int(input('Digite un número: '))
    
    for i in range(1, numero):
        if numero % i == 0:
           cont += 1
    
    if cont > 2:
        print('El número {} no es primo.'.format(numero))
    else:
        print('El número {} es primo.'.format(numero))
        
if __name__ == '__main__':
    main()